python/triton/compiler/compiler.py

```diff
@@ -309,14 +310,15 @@ def make_hash(fn, arch, env_vars, **kwargs):
         num_ctas = kwargs.get("num_ctas", 1)
         num_stages = kwargs.get("num_stages", 3)
         waves_per_eu = kwargs.get("waves_per_eu", 0)
+        matrix_instr_nonkdim = kwargs.get("matrix_instr_nonkdim", 0);
```

Comment:
The only benefit to add them explicitly is that we can still tune them even they are not explicitly defined as kernel arguments.

Reply:
When I added `pre_load_v` as a tuning parameter, I just added it in the config of the autotuner and kernel argument as a `tl.constexpr`. Nothing is changed in the compiler.py. And it is treated as `BLOCK_M` instead of `num_warps`.

It seems that there are two kinds of kernel arguments: one is meta-parameters like BLOCK_M, the other is compilation options like num_warps according to the explanation here: https://github.com/ROCmSoftwarePlatform/triton/blob/461d72e5477d1659dc05e10060db4db3907c958f/python/tutorials/03-matrix-multiplication.py#L162
And the only difference between to two is whether we can set default values to them. For meta-parameters, if nothing is set, there will be an error like `missing 1 required positional argument: 'pre_load_v'`.

P.s. I think both kinds are compilation options, since the kernel needs to be recompiled if the values is changed.

Reply:
This is interesting approach, it definitely worth to try it.

I have only one concern about it.
User have to declare this constant by itself, and if he/she make a mistake, this mistake will not be reported.

For example, we use `MATRIX_INSTR_NONKDIM` constant to control MFMA behavior, user can write this code:

```
@triton.jit
def kernel(MTRIX_INSTR_NONKDIM: tl.constexpr):
    ...

kernel[grid](MTRIX_INSTR_NONKDIM = 16)
```

This code is correct in therm of a language, but it does not do what we want and it mistake is not reported.

Reply:
I am thinking, maybe we can introduce some additional decorator to pass AMD specific options to kernel without messing with upstream interfaces.

Reply:
I see your point. 
`MATRIX_INSTR_NONKDIM` and `waves_per_eu` are needed explicitly in the lowering passes. However, `pre_load_v` and `BLOCK_M` are only needed in the python level frontend, so the `compile()` function don't care about them.

Reply:
Upstream has added a lot of hopper specific parameters to the list already, which is far from clean at all.
I agree with you that we should have a "bag" for AMD options. And we should also suggest to upstream to put all these NVIDIA parameters into another bag.